### ISSUE TITLE
Unleash isEnabled always return default value when minifyEnable true

### ISSUE THREAD

**Issue by fahmisdk6** [Unleash/unleash-android-proxy-sdk#22]
### Describe the bug

I have implemented Unleash in my project, when running the project with minifyEnable false the toggles work correctly. 
But when running the project with minifyEnable true the toggles always return default value.

My gradle 

```
release {
    minifyEnabled true
    proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
}
```

My Proguard
```
-keep class io.getunleash.** { *; }
```

or 
```
-keep public class io.getunleash.** { *; }
```

I'm pretty sure the connection with unleash-proxy is working because i log the api calls to chuck, but somehow getting the value using `client.isEnabled(toggle-name)` is always return false.

### Steps to reproduce the bug

1. Implement unleash sdk
2. Enable proguard
3. Run project
4. Get Toggle using `client.isEnabled("toggle-name")`

### Expected behavior

toggle return correct value based on API and Dashboard

### Screenshots

![image](https://user-images.githubusercontent.com/10940190/165234985-b683956b-72aa-4c9e-8f82-60d344243818.png)

### Unleash version

0.4.0

### Hosting type

Self-hosted

**Comment by chriswk** [comment]
Thanks @fahmisdk6 for a great bug report. I will have a look today and confirm that I can reproduce it, and then see if I can't find a fix. I'll keep you posted

**Comment by chriswk** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Android SDK with minify enabled always returns false](https://trello.com/c/e07cal95/1087-android-sdk-with-minify-enabled-always-returns-false)

**Comment by chriswk** [comment]
I've been trying to reproduce this. PR #23 gives a sample that should be possible to test in release mode.
It receives updates in both debug and release mode.

The only difference between your two test runs @fahmisdk6  is that minify is enabled?

**Comment by fahmisdk6** [comment]
Yes the only difference is minifyEnable, but i'll check again and look into your sample.
Thank you for checking @chriswk

**Comment by chriswk** [comment]
If you still run into problems, I can push a 0.4.1 with a bit more logging at debug level.

**Comment by Arm63** [comment]
Add  classpath 'com.android.tools:r8:3.3.28' to your build.gradle (project)  file. This solved my problem 
(https://issuetracker.google.com/issues/206855609#comment9)

Additional info 
    ext.kotlin_version = '1.5.30'
    classpath 'com.android.tools.build:gradle:7.0.2
    
    minifyEnabled true
    
    -keep class kotlin.Metadata { *; }
    -keep public class io.getunleash.** {*;}

**Comment by ilya842** [comment]
I have a similar problem. Debug application is working fine, but when I assemble `release` build, unleash stops working. I tried to debug this problem. Exception happens when jackson library tries to deserialize json. 

Error output (UnleashFethcer.kt, line 84):  
`com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Cannot construct instance of "io.getunleash.data.ProxyResponse" (no Creators, like default constructor, exist): cannot deserialize from Object value (no delegate- or property-based Creator)`

I have this proguard rules
`-keep public class io.getunleash.** {*;}`

kotlin version: 1.5.30
gradle version: 7.0.2
unleash version: 0.4.0

**Comment by ilya842** [comment]
@Arm63 thank you for your tip. It helped for me.

**Comment by fahmisdk6** [comment]
Hi all, since there is solution from @Arm63, i think we can close the issue

**Comment by jagtejsodhi** [comment]
Hi all, this is still an issue on my end. I tried @Arm63's workaround fix, but didn't work. After reviewing @ilya842 error output, I figured the issue was caused by proguard stripping out a part of the Jackson library. Adding this proguard rule fixed the issue.

```
-keep class com.fasterxml.** { *; }
```

@fahmisdk6 When an issue as major as this comes up, I would highly suggest that a legitimate fix be made in the Unleash code base versus just closing the issue when a workaround is found. If Unleash can't fix the issue immediately, I would suggest adding a "Known Issues and Workarounds" section to the readme and documentation.
